week2/ex3_done.py

```diff
@@ -0,0 +1,104 @@
+# 3.	Elements counting. It refers to the sections 1.12 and 1.4-1.5 of the Python Cookbook. Implement the Counter class. It should only support operations like __iter__, __getitem__ and .most_common(n). Try to make it as efficient as possible in terms of performance. Think of an underlying data structure that should be used and also a base class.
+import collections
+import collections.abc
+
+
+def _identity(s):
+    """"
+    Default mutator function.
+    """
+    return s
+
+
+class CustomCounter(collections.abc.MutableMapping):
+    """
+    Overrides the 5 methods of a MutableMapping:
+    __getitem__, __setitem__, __delitem__, __iter__, __len__
+
+    ...and the 3 non-Mapping methods of Counter:
+    elements, most_common, subtract
+    """
+
+    def __init__(self, values=None, *, mutator=_identity):
+        self._mutator = mutator
+        if values is None:
+            self._counter = collections.Counter()
```

Comment:
The tasks asks to implement your own Counter class. Not to wrap around the standard one. 
Your code should have no references to `collections.Counter`